pkg/controller/lifecycle/actuator.go

```diff
@@ -115,11 +114,9 @@ func (a *actuator) Reconcile(ctx context.Context, ex *extensionsv1alpha1.Extensi
 	// Shoots that don't specify a DNS domain or that are scheduled to a seed that is tainted with "DNS disabled"
 	// don't get an DNS service
 
-	// TODO: remove the deprecated taint check in a future version
 	if !seedSettingShootDNSEnabled(cluster.Seed.Spec.Settings) ||
-		gardencorev1beta1helper.TaintsHave(cluster.Seed.Spec.Taints, gardencorev1beta1.DeprecatedSeedTaintDisableDNS) ||
 		cluster.Shoot.Spec.DNS == nil {
-		a.Info("DNS domain is not specified, the seed .spec.settings.shootDNS.enabled=false or the seed is tainted with 'disable-dns', therefore no shoot dns service is installed", "shoot", ex.Namespace)
+		a.Info("DNS domain is not specified, the seed .spec.settings.shootDNS.enabled=false, therefore no shoot dns service is installed", "shoot", ex.Namespace)
```

Comment:
@ialidzhikov Is this the correct way to remove the deprecated taint check?

Reply:
Yes, looks good to me, we should rely on Seed's `.spec.settings.shootDNS.enabled` field. With https://github.com/gardener/gardener/pull/2955 the `seed.gardener.cloud/disable-dns` taint is removed.